dht_net.go

```diff
@@ -66,32 +70,44 @@ func (dht *IpfsDHT) handleNewMessage(s inet.Stream) {
 		case nil:
 		default:
 			s.Reset()
-			log.Debugf("Error unmarshaling data: %s", err)
+			log.LogKV(ctx,
+				"error", err,
+				"when", "unmarshaling request",
+			)
+			log.SetTag(ctx, "error", true)
 			return
 		}
 
+		log.LogKV(ctx,
```

Comment:
so these will keep accumulating while the loop is running and not actually logging until the span is done.... Not good.

Reply:
Maybe I am just misunderstanding how the new logger works though.